green_cli/tx.py

```diff
@@ -0,0 +1,207 @@
+        tx.setdefault('addressees', [])
+        tx['addressees'].extend(details['addressees'])
+
+@tx.command()
+def raw():
+    """Get the raw transaction hex"""
+    click.echo(_load_tx(allow_errors=False)['transaction'])
+
+@tx.command()
+def dump():
+    """Dump the full transaction json representation"""
+    click.echo(json.dumps(_load_tx(allow_errors=True)))
+
+@tx.command()
+@click.argument('tx_json', type=click.File('r'))
+@with_login
+def load(session, tx_json):
+    """Load a transaction from json, see also dump"""
+    raw_tx = tx_json.read()
+    tx = json.loads(raw_tx)
+    _save_tx(_create_tx(tx))
+
+@tx.command()
+def info():
+    """Show summary information about the current tx"""
+    tx = _load_tx(allow_errors=True)
+    if tx['error']:
+        click.echo(f"ERROR: {tx['error']}")
+
+    if 'txhash' in tx:
+        click.echo(f"txhash: {tx['txhash']}")
+    click.echo(f"user signed: {tx['user_signed']}")
+    click.echo(f"server signed: {tx['server_signed']}")
+    for addressee in tx['addressees']:
+        click.echo(f"output: {addressee['address']} {addressee['satoshi']}")
+    for asset, amount in tx['satoshi'].items():
+        click.echo(f"total {asset}: {amount}")
+    click.echo(f"fee: {tx['fee']} sat")
+    click.echo(f"fee rate: {tx['calculated_fee_rate']} sat/kb")
+
+@tx.group(name='coin')
+def tx_coin():
+    """Coin selection"""
+    pass
+
+@tx_coin.command()
+@with_login
+def status(session):
+    """Show status of coins for current tx"""
+    tx = _load_tx(allow_errors=True)
+
+    selected = 0
+    click.echo("selected:")
+    for utxo in tx['used_utxos']:
+        confs = confs_str(utxo['block_height'])
+        click.echo(f"\t{utxo['satoshi']} {utxo['address_type']} {confs} {utxo['txhash']}:{utxo['pt_idx']}")
+        selected += utxo['satoshi']
+    click.echo(f"\ttotal: {selected}")
+
+    available = 0
+    click.echo("available:")
+    for asset, utxos in tx['utxos'].items():
+        for utxo in utxos:
+            confs = confs_str(utxo['block_height'])
+            click.echo(f"\t{utxo['satoshi']} {utxo['address_type']} {confs} {utxo['txhash']}:{utxo['pt_idx']}")
+            available += utxo['satoshi']
+        click.echo(f"\ttotal: {available}")
+
+def _filter_utxos(utxo_filter, utxos):
+    txhash, _, vout = utxo_filter.partition(':')
+    selected = []
+    for utxo in utxos:
+        if txhash == '*' or txhash == utxo['txhash']:
+            if not vout or vout == '*' or int(vout) == utxo['pt_idx']:
+                selected.append(utxo)
+    return selected
+
+@tx_coin.command()
+@with_login
+def auto(session):
+    """Enable automatic coin selection"""
+    with Tx(allow_errors=True) as tx:
+        tx['utxo_strategy'] = 'default'
+
+@tx_coin.command()
+@click.argument('utxo')
+@with_login
+def select(session, utxo):
+    """Select coins/utxos"""
+    with Tx(allow_errors=True) as tx:
+        tx['utxo_strategy'] = 'manual'
+        for utxo in _filter_utxos(utxo, tx['utxos']['btc']):
```

Comment:
should we allow for other assets as well?

Reply:
For now I have left this disabled for liquid as there are some issues that need to be resolved, so that can be in a future PR